modules/gapi/src/backends/common/gbackend.hpp

```diff
@@ -58,32 +58,37 @@ namespace gimpl {
     struct Data;
     struct RcDesc;
 
-    struct GAPI_EXPORTS RMatMediaAdapterBGR final: public cv::RMat::Adapter
+    struct GAPI_EXPORTS RMatMediaFrameAdapter final: public cv::RMat::Adapter
     {
-        explicit RMatMediaAdapterBGR(const cv::MediaFrame& frame) : m_frame(frame) { };
+        explicit RMatMediaFrameAdapter(const cv::MediaFrame& frame,
+            std::function<cv::Mat(const GFrameDesc&, const cv::MediaFrame::View&)> frameViewToMat) :
```

Comment:
This looks weird, why do we need a callback like this here?

Reply:
Got the idea after reviewing the rest of this code. I'd recommend to give this `std::function` a name (right within this class). See how our kernels are implemented in e.g. Fluid backend.

Reply:
Fixed